src/systems/inter_mesh_projection.C

```diff
@@ -38,27 +49,144 @@ namespace libMesh
 
     void InterMeshProjection::project_system_vectors()
     {
-        // Number of vectors to be projected
-        unsigned int n_vecs = from_system.n_vectors();
-        libmesh_assert_equal_to (to_system.n_vectors(), n_vecs);
+        libmesh_assert_equal_to (to_system.n_vectors(), from_system.n_vectors() );
 
         // Number of variable components in each vector
         unsigned int n_vars = from_system.n_vars();
         libmesh_assert_equal_to (to_system.n_vars(), n_vars);
 
+        // We are going to use the multi-variable MeshFunction, so we can pass
+        // a single vector of variables rather than have a MeshFunction for each variable
+	    std::vector<unsigned int> variables_vector;
+
+        for (unsigned int j = 0; j != n_vars; ++j)
+        {
+          variables_vector.push_back(j);
+        }
+
         // Update system vectors with contributions from all processors
         from_system.update();
 
-        // Loop over all the vectors in the system
-        for (unsigned int i = 0; i != n_vecs; ++i)
-        {
+        // Any system holds the solution along with the other vectors system.vectors
+        // We will first project the solution and then move to the system.vectors
+
+        // Construct local version of the current system vector
+	    // This has to be a serial vector
```

Comment:
Technically it just has to be a ghosted vector whose algebraically ghosted values cover a domain which is a superset of the to-system's domain ... that's hard to do and we can skip it now, but we might want to put a "FIXME" comment or something to that effect so we don't simply forget about the problem until the poor scalability bites someone.

Reply:
Added in upcoming commit.